database-migration/src/main/java/uk/ac/ebi/eva/dbmigration/mongodb/MongobeeHelper.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.dbmigration.mongodb;
+
+import com.github.mongobee.Mongobee;
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientOptions;
+import com.mongodb.MongoCredential;
+import com.mongodb.ReadPreference;
+import com.mongodb.ServerAddress;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class MongobeeHelper {
+
+    public static Mongobee buildMongobee(DatabaseParameters databaseParameters) {
+        List<ServerAddress> servers = getServers(databaseParameters);
+        List<MongoCredential> credentials = getCredentials(databaseParameters);
+        MongoClientOptions options = getOptions(databaseParameters);
+
+        Mongobee mongobee = new Mongobee(new MongoClient(servers, credentials, options));
+        mongobee.setDbName(databaseParameters.getDbName());
+        return mongobee;
+    }
+
+    private static List<ServerAddress> getServers(DatabaseParameters databaseParameters) {
+        List<ServerAddress> addresses = new ArrayList<>();
+
+        int port = ServerAddress.defaultPort();
+        if (databaseParameters.getDbPort() != null) {
+            port = Integer.parseInt(databaseParameters.getDbPort());
+        }
+
+        String hosts = databaseParameters.getDbHosts();
+        if (hosts == null) {
+            hosts = ServerAddress.defaultHost();
+        }
+
+        for (String host : hosts.split(",")) {
+            addresses.add(new ServerAddress(host, port));
+        }
+
+        return addresses;
+    }
+
+    private static List<MongoCredential> getCredentials(DatabaseParameters databaseParameters) {
```

Comment:
I think it only complains if user or password are null, but it is fine if the authentication DB is not. At least from a Mongo perspective, without considering the Java API, it is perfectly fine not to use an auth DB (see local testing databases).